### DISCUSSION TITLE
Home Assistant Integration

### DISCUSSION THREAD

**Discussion by soloam** [firefly-iii/firefly-iii#8757] [General]
Hello, I'm making a Home Assistant Integration to pull data from firefly into Home Assistant. I would like to ask if it's OK to use names and logos from firefly in the integration! I'll make links and references in the repo to this repo!

Thank you

**Comment by JC5** [comment]
I thought there already is a Home Assistant integration?

**Reply by soloam** [reply]
On the core integration and on the Hacs (community store) it does not exist! Also on Google I only can find people playing around with the api, no native integration.

**Reply by soloam** [reply]
I think you mean the home assistant addon! That is to run firefly on the home assistant machine... But no data is send to home assistant... For that you need a integration

**Reply by lucaslgr** [reply]
I'm just a curious here. But what kind of automatization you could have integrating Firefly with HA ? 🤔

**Reply by soloam** [reply]
Basicly with the information on HA you can do whatever you can think of... Invoices due date notifications, budget control, account control! You can use HA to deliver information with other presentation! You can use home assistant voice assistant! You can use Home Assistant automation to create or edit transactions!...

**Reply by lucaslgr** [reply]
Very interesting, now I got it! thank you!!

**Reply by soloam** [reply]
I'm looking for beta testers to help me target bugs and potencial features prior to release on HACS. If you would like to test it out, or anyone else, say something so that I can provide the link to add to HACS custom repositories!

**Reply by diamant-x** [reply]
I would do it happily, but only have one instance of FF3, with nearly a decade of data working with so can't allow myself to loose it so I won't risk it. If FF3 had some easy way to export-import data to a new instance I would do it happily (that would also help to migrate from one DB engine to another, say moving from a MariaDB i'm sharing with other apps, to a SQLite instance so i can back it up in a single file.
Sorry!

**Reply by soloam** [reply]
The interface only reads data, it does not write! The only way to the interface to write is with the services that will allow to create transfers and bills!

**Reply by soloam** [reply]
The first version of the integration is on my github! You can add it to HACS custom repositories as integration and after reboot you can install it on the UI!

Please note that I’ll not be retrofitting configs to versions until its final! So on some releases of beta I’ll add to the release notes a information to delete the integration and add it again. No need to remove from HACS, just remove from Home Assistant and make the config again!

At the moment I have accounts and categories working. Feedbacks are more than welcome, also feature requests. But do it on the github issues of my repo, not here!

Please expect bugs, this is a very early Beta!

https://github.com/soloam/ha-fireflyiii-integration

**Comment by JC5** [comment]
Ah yes, I see.

In that case, please read: https://docs.firefly-iii.org/explanation/more-information/logo/

You have my permission to use the logo for the context of this integration, and I would suggest a name like "HA integration for Firefly III".

**Reply by soloam** [reply]
Thank you! Before I submit it to the store I'll show you for you approval!

**Comment by diamant-x** [comment]
Just my two cents here. FF3 has webhooks option (although i haven't read yet how it works) and Home Assistant supports automations triggered by webhooks calls.
That would allow for automations based on FF3 transactions.

But if you just mean sensors for balances and such then I guess an integration is needed yes (or manual sensor with the API and the REST integration).

Just couple ideas that may work with HA as-is. Haven't thought of any use case for FF3 in HA though. Maybe I can think of leveraging the HA rule engine (Which is veery strong) to handle complex scenarios in FF3 through the API.

**Reply by soloam** [reply]
The idea is to bring Accounts, Budgets, Invoices, Piggy Banks, and Categories into Home Assistant as sensors, then you can use in automations, also services to create and edit transactions, this would allow to do just what you say... use Home Assistant to process more complex scenarios. At the moment I do some complex logic with nodered, but with a integration like this, it would allow Home Assistant Users to do some automations to edit or notify.

**Comment by soloam** [comment]
@JC5 when I pull from the api of categories I can get the final value of that category (the spent + the earned). Whats the best way to get the same information to "No Category"? I'm using search to get all transactions without category, and doing the math! Is there a better way to get that?

Tks

**Comment by soloam** [comment]
The integration is ready to push to HACS (Home Assistant Integration Store), I'm just waiting for approval from Home Assistant. I added references to fireflyiii in the readme, but stated that the project was independent from the FireflyIII app.

I have to improve the readme with more information, but for now it's ok to publish:
https://github.com/soloam/ha-fireflyiii-integration

Thank you all that supported the tests in the beta.

**Reply by JC5** [reply]
Very cool! Thanks for building this! I will add it to the documentation for sure!

**Reply by soloam** [reply]
Tks @JC5 . It was only possible because of your excellent work with the app and api! Thank you for adding to your documentation, it will help people know that the integration is available!

Best regards

**Comment by poudenes** [comment]
Hey Guys, 

They FFIII got lately a big change and update. The integration for HA is broken. Maybe you can help each other to make this integration working again :D ... im sorry im not a programmer so I can't help :(
